app/components/layout/keyboard_layout/keyboard_layout.js

```diff
@@ -41,52 +40,36 @@ export default class KeyboardLayout extends PureComponent {
     }
 
     onKeyboardWillHide = () => {
-        this.setState({bottom: 0});
+        this.setState({
+            keyboardHeight: 0,
+        });
     };
 
-    onKeyboardChange = (e) => {
-        if (!e) {
-            this.setState({bottom: 0});
-            return;
-        }
-
-        const {endCoordinates} = e;
-        const {height} = endCoordinates;
-
-        this.setState({bottom: height});
+    onKeyboardWillShow = (e) => {
+        this.setState({
+            keyboardHeight: e?.endCoordinates?.height || 0,
```

Comment:
The new version of RN also supports this operator. I forget the JS name for it, but it's called the safe navigation operator in Groovy. It is just a proposal right now and NOT PART OF JS at this point, so let me know what you think about using it.

If we wanted to support it in the other libraries, we'd need the Babel plugin that RN uses